### PR TITLE
Adding Data Source Forwarding Rules

### PR DESCRIPTION
**Pull request by jesse-l**
Adding a new data source that will allow users to list forwarding rules that they have inside a specific region of a Google Cloud Project. This PR is for [#17294](https://github.com/hashicorp/terraform-provider-google/issues/17294)

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-datasource
google_compute_forwarding_rules
```

### CODE REVIEW

File: mmv1/third_party/terraform/services/compute/data_source_google_compute_forwarding_rules.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"rules": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeMap},
```
Comment (c2thorn) [inline mmv1/third_party/terraform/services/compute/data_source_google_compute_forwarding_rules.go:30]: why not reuse the resource schema of the existing forwarding rule schema? similar to https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/services/sql/data_source_sql_databases.go#L31

I'm not sure a string map will be the best experience for users.

File: mmv1/third_party/terraform/services/compute/data_source_google_compute_forwarding_rules_test.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	poolName := fmt.Sprintf("tf-%s", acctest.RandString(t, 10))
+	ruleName := fmt.Sprintf("tf-%s", acctest.RandString(t, 10))
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceGoogleForwardingRuleConfig(poolName, ruleName),
+				Check:  acctest.CheckDataSourceStateMatchesResourceStateWithIgnores("data.google_compute_forwarding_rule.my_forwarding_rule", "google_compute_forwarding_rule.foobar-fr", map[string]struct{}{"port_range": {}, "target": {}}),
```
Comment (c2thorn) [inline mmv1/third_party/terraform/services/compute/data_source_google_compute_forwarding_rules_test.go:23]: what's the reasoning for ignoring fields here?

**Review by c2thorn — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/services/compute/data_source_google_compute_forwarding_rules.go (hunk 1 above)
Comment (jesse-l) [reply mmv1/third_party/terraform/services/compute/data_source_google_compute_forwarding_rules.go]: I have updated the return type to be the Forwarding Rules schema now.

File: mmv1/third_party/terraform/services/compute/data_source_google_compute_forwarding_rules_test.go (hunk 2 above)
Comment (c2thorn) [reply mmv1/third_party/terraform/services/compute/data_source_google_compute_forwarding_rules_test.go:23]: @jesse-l why should we ignore here? Are we expecting it to not match on purpose?

**Review by c2thorn — APPROVED** [review]

**Pull request merged**